ethers-contract/src/call.rs

```diff
@@ -54,12 +56,16 @@ pub enum ContractError<M: Middleware> {
     DetokenizationError(#[from] InvalidOutputType),
 
     /// Thrown when a middleware call fails
-    #[error("{0}")]
-    MiddlewareError(M::Error),
+    #[error("{e}")]
+    MiddlewareError { e: M::Error },
```

Comment:
deliberately broken 3rd party direct instantiation, so that users must use the `from_middleware_error` method

similar for provider error. broke it to force `From` usage